### PR TITLE
FAQ - Better QR code entries

### PR DESCRIPTION
**Pull request by heinezen**
Fixes https://github.com/corona-warn-app/cwa-website/issues/549

What has been done:
* Split the entry into two: One for scanning, one for a delayed test result
* Restructure the `qr_scan` entry for better readability
* Remove notice about entering the code twice, Should be covered by https://www.coronawarn.app/de/faq/#test_multiple_devices

### CODE REVIEW

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
There is a basic mistake in the section
"I get an error while scanning the QR code. What can I do?"
referring to the reason 
"The test result was not allowed to be transmitted."

That is not the way it works.

As @thomasaugsten noted in https://github.com/corona-warn-app/cwa-app-android/issues/1551#issuecomment-723227499 a user should normally be able to scan in a correct QR code. This is independent of whether the consent field on the form is marked, or if the lab has entered the code into the lab system. 

The reasons it might not work are:
- defective QR code on printed form
- the code has already been used
and the reasons for this include:
- the form has been copied and re-used for multiple tests
- the QR code has already been scanned in by the user themselves and then deleted

The error message `submission_qr_code_scan_invalid_dialog_body` does not distinguish between the different error causes.

<string name="submission_qr_code_scan_invalid_dialog_headline">"**QR code is invalid**"</string>
<string name="submission_qr_code_scan_invalid_dialog_body">"The QR code is invalid or has been registered on another smartphone already. You will receive your test result from the test center or laboratory regardless of the validity of the QR code. If you are diagnosed with COVID-19, the public health authority will be notified through the legally prescribed channel and will contact you."</string>
   

So this reason (missing consent and missing lab input) needs to be moved into the section
"I have scanned a QR code, but the test result could not be retrieved for days."

**Comment by heinezen** [comment]
New revised entry is here.

File: src/data/faq.json [hunk 1]
```diff
@@ -100,15 +100,29 @@
...
-                            "We are currently looking into how to improve the app's user interface to detect QR Codes that are not working and inform you about them. You can find more details about the errors <a href='https://github.com/corona-warn-app/cwa-documentation/issues/400' target='_blank' rel='noopener noreferrer'>on GitHub</a>."
+                            "If you have scanned your QR code into the App after a corona test and do not receive a test result via the App within a few days, it is possible that the respective laboratory is not yet fully cooperating with the Corona-Warn-App or the field 9 'Consent of the insured person to transmit the test result for the purpose of the Corona-Warn-App' has not been ticked in the form.",
+                            "Please check whether the laboratory with which the doctor or the test center works is connected to the Corona-Warn-App and whether field 9 is checked on the form. Only then is it ensured that the procedure used works with the Corona-Warn-App."
```
Comment (MikeMcC399) [inline src/data/faq.json:122]: comma is missing at end of line, leading to a syntax error.

"Please check whether the laboratory with which the doctor or the test center works is connected to the Corona-Warn-App and whether field 9 is checked on the form. Only then is it ensured that the procedure used works with the Corona-Warn-App."

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Syntax error.

If you could correct this, it will be easier to review using the local webserver.

File: src/data/faq.json [hunk 2]
```diff
@@ -100,15 +100,29 @@
...
-                            "If you have already scanned the QR Code, it unfortunately cannot be scanned a second time, nor is it possible to delete a previously entered QR Code and scan it in again. We have to make sure that only one phone can scan the QR Code and transmit its diagnostic keys. For privacy reasons, we cannot store the phone's ID, so the QR Code is only valid for one scan.",
-                            "We are currently looking into how to improve the app's user interface to detect QR Codes that are not working and inform you about them. You can find more details about the errors <a href='https://github.com/corona-warn-app/cwa-documentation/issues/400' target='_blank' rel='noopener noreferrer'>on GitHub</a>."
+                            "If you have scanned your QR code into the App after a corona test and do not receive a test result via the App within a few days, it is possible that the respective laboratory is not yet fully cooperating with the Corona-Warn-App or the field 9 'Consent of the insured person to transmit the test result for the purpose of the Corona-Warn-App' has not been ticked in the form.",
```
Comment (MikeMcC399) [inline src/data/faq.json:121]: Capitalise corona -> Corona

File: src/data/faq_de.json [hunk 3]
```diff
@@ -100,15 +100,30 @@
...
                     },
                     {
-                        "title": "Probleme mit dem QR-Code",
+                        "title": "Beim Einscannen erhalte des QR Codes ich einen Fehler. Was kann ich tun?",
+                        "anchor": "qr_scan",
+                        "active": true,
+                        "textblock": [
+                            "Dieses Problem kann zwei verschiedene Ursachen haben:",
+                            "<b>Der QR Code wurde bereits einmal zur Registrierung benutzt</b>",
+                            "Jeder QR Code den sie erhalten kann nur einmal in der App eingegeben werden. Falls Sie die Corona-Warn-App auf mehreren Geräten verwenden, können Sie den QR Code nur in einem der Geräte registrieren.",
```
Comment (MikeMcC399) [inline src/data/faq_de.json:109]: erhalten kann -> erhalten, kann  (missing comma)

**Comment by MikeMcC399** [comment]
There is new information in https://github.com/corona-warn-app/cwa-app-android/issues/1551#issuecomment-745266898 from @dsarkar who wrote:

> It is now possible to scan the QR code twice

Does that mean a QR code can be scanned once or twice, but not three or more times?

This needs some clarification.

**Comment by heinezen** [comment]
@MikeMcC399 I fixed the minor errors.

Regarding scanning the QR code twice: This should be a recent change. I'll have to check how this manifests itself when using the app.

----
Corona-Warn-App Open Source Team

File: src/data/faq.json [hunk 4]
```diff
@@ -100,15 +100,29 @@
...
+                            "Please check whether the laboratory with which the doctor or the test center works is connected to the Corona-Warn-App and whether field 9 is checked on the form. Only then is it ensured that the procedure used works with the Corona-Warn-App.",
+                            "Currently, 90% of the existing test capacities in Germany are covered by the complete digital process. We are continuing to work with the laboratories to close the gaps and to connect and upgrade all laboratories for the process. Therefore it can happen that a laboratory gets the order for the test with the correct form, but has to transmit the test result to the tested person in another way.",
+                            "In this case, please call the responsible medical staff or test center/health authority to get your test result. If you are informed of a positive test result, you can request a TeleTAN by calling the hotline at +49 800 7540002 to warn your risk encounters. Please enter the TeleTAN into the Corona Warn App by following these steps:",
+                            "<ul><li>'Inform & Help' -> 'Continue' -> 'TAN'</li></ul>"
```
Comment (MikeMcC399) [inline src/data/faq.json:125]: ` "<ul><li>'Inform & Help' -> 'Continue' -> 'TAN'</li></ul>"` does not correspond to UI
Change to
` "<ul><li>'LEARN MORE AND HELP' -> 'NEXT' -> 'TAN'</li></ul>"`

---
LEARN MORE AND HELP

---
NEXT

---
TAN_en

File: src/data/faq_de.json [hunk 5]
```diff
@@ -100,15 +100,30 @@
...
+                            "Derzeit sind 90% der vorhandenen Testkapazitäten in Deutschland durch den kompletten digitalen Prozess abgedeckt. Wir arbeiten mit den Laboren weiter daran, die Lücken zu schließen und alle Labore für den Prozess anzubinden und aufzurüsten. Deshalb kann es vorkommen, dass ein Labor zwar die Beauftragung für den Test mit dem richtigen Formular bekommt, das Testergebnis aber auf einem anderen Weg an die getestete Person übermitteln muss.",
+                            "Rufen Sie bitte beim zuständigen ärztlichen Fachpersonal bzw. Testcenter/Gesundheitsamt an, um Ihr Testergebnis zu erhalten. Wird Ihnen dabei ein positives Testergebnis mitgeteilt, können Sie über die Hotline unter der +49 800 7540002 eine TeleTAN beantragen, um über die Corona-Warn-App Ihre Risikobegegnungen zu warnen. Tragen Sie für diesen Zweck die erhaltene TeleTAN unter",
+                            "<ul><li>'Informieren & Helfen' -> 'Weiter' -> 'TAN'</li></ul>",
```
Comment (MikeMcC399) [inline src/data/faq_de.json:125]: `"<ul><li>'Informieren & Helfen' -> 'Weiter' -> 'TAN'</li></ul>",`
capitalise to correspond to appearance in UI
`"<ul><li>'INFORMIEREN & HELFEN' -> 'WEITER' -> 'TAN'</li></ul>",`

---
INFORMIEREN   HELFEN

---
WEITER

---
TAN_de

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
The instructions for submitting a TAN through the English UI are not correct, so I suggested corrections together with screen shots for verification.

For consistency I also suggested capitalising the words for the German UI so that they are exactly the same as they appear on the user's screen.

**Comment by MikeMcC399** [comment]
@heinezen 
> Regarding scanning the QR code twice: This should be a recent change. I'll have to check how this manifests itself when using the app.

You can disregard this point, as it was a typo from @dsarkar !  
I couldn't imagine how this would work and he confirmed the mistake in https://github.com/corona-warn-app/cwa-app-android/issues/1551#issuecomment-745356579.

**Comment by MikeMcC399** [comment]
According to https://www.coronawarn.app/en/blog/2020-12-16-corona-warn-app-version-1-9/ CWA 1.9 is being released to all devices over the next 48 hours, and the screen shots there are different.

To enter a TAN:
- English:
NEXT STEPS > Enter TAN

- German;
NÄCHSTE SCHRITTE > TAN-Eingabe

I can only test with the release/1.9.x debug version at the moment. I don't have access to any released 1.9.x version on Android.

**Comment by heinezen** [comment]
@MikeMcC399 I finally got the texts from the new release. They went trough several iterations.

The font should noot be apitalized anymore in 1.9

**Comment by MikeMcC399** [comment]
@heinezen  
> The font should noot be apitalized anymore in 1.9

Did you check to see how the text is displayed on a device? I am seeing the text "NEXT STEPS" / "NÄCHSTE SCHRITTE" displayed in capital letters.

Also, in the app version I'm running 1.9.1-RC1, cloned from the "main" branch, the step 'Next' / 'Weiter' is no longer shown - so one step less than in the version 1.7.1.

**Comment by heinezen** [comment]
@MikeMcC399 I'll wait for the Android release then, because I presume the mockups I looked at are outdated. Development is too rapid in this case 🔧 :D 

The Android release should be completely rolled out in the next days

File: src/data/faq.json [hunk 6]
```diff
@@ -100,15 +100,29 @@
...
+                            "Please check whether the laboratory with which the doctor or the test center works is connected to the Corona-Warn-App and whether field 9 is checked on the form. Only then is it ensured that the procedure used works with the Corona-Warn-App.",
+                            "Currently, 90% of the existing test capacities in Germany are covered by the complete digital process. We are continuing to work with the laboratories to close the gaps and to connect and upgrade all laboratories for the process. Therefore it can happen that a laboratory gets the order for the test with the correct form, but has to transmit the test result to the tested person in another way.",
+                            "In this case, please call the responsible medical staff or test center/health authority to get your test result. If you are informed of a positive test result, you can request a TeleTAN by calling the hotline at +49 800 7540002 to warn your risk encounters. Please enter the TeleTAN into the Corona Warn App by following these steps:",
+                            "<ul><li>'Next Steps' -> 'Next' -> 'Enter TAN'</li></ul>"
```
Comment (MikeMcC399) [inline src/data/faq.json:125]: Change `"<ul><li>'Next Steps' -> 'Next' -> 'Enter TAN'</li></ul>"` to
`"<ul><li>'NEXT STEPS' -> 'Enter TAN'</li></ul>"`

Screenshots from released CWA Android 1.9.1
---
En_have_you_been_tested

---
En_did_you_test_positive

File: src/data/faq_de.json [hunk 7]
```diff
@@ -100,15 +100,30 @@
...
+                            "Derzeit sind 90% der vorhandenen Testkapazitäten in Deutschland durch den kompletten digitalen Prozess abgedeckt. Wir arbeiten mit den Laboren weiter daran, die Lücken zu schließen und alle Labore für den Prozess anzubinden und aufzurüsten. Deshalb kann es vorkommen, dass ein Labor zwar die Beauftragung für den Test mit dem richtigen Formular bekommt, das Testergebnis aber auf einem anderen Weg an die getestete Person übermitteln muss.",
+                            "Rufen Sie bitte beim zuständigen ärztlichen Fachpersonal bzw. Testcenter/Gesundheitsamt an, um Ihr Testergebnis zu erhalten. Wird Ihnen dabei ein positives Testergebnis mitgeteilt, können Sie über die Hotline unter der +49 800 7540002 eine TeleTAN beantragen, um über die Corona-Warn-App Ihre Risikobegegnungen zu warnen. Tragen Sie für diesen Zweck die erhaltene TeleTAN unter",
+                            "<ul><li>'Nächste Schritte' -> 'Weiter' -> 'TAN-Eingabe'</li></ul>",
```
Comment (MikeMcC399) [inline src/data/faq_de.json:125]: Change `"<ul><li>'Nächste Schritte' -> 'Weiter' -> 'TAN-Eingabe'</li></ul>",` to
`"<ul><li>'NÄCHSTE SCHRITTE' -> 'TAN-Eingabe'</li></ul>",`

Screenshots from released CWA Android 1.9.1
---
De_wurden_Sie_gestestet

---
De_Ihr_Test_ist_positiv

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Comments are updated with screen shots from the released version of CWA Android 1.9.1 obtained from the Google Play Store.

**Comment by heinezen** [comment]
Updated to use capitalization.

**Comment by heinezen** [comment]
And merged

**Pull request merged**